camel/societies/babyagi_playing.py

```diff
@@ -0,0 +1,242 @@
+            pass to the task specify agent. (default: :obj:`None`)
+        task_create_agent_kwargs (Dict, optional): Additional arguments to
+            pass to the task create agent. (default: :obj:`None`)
+        task_prioritize_agent_kwargs (Dict, optional): Additional arguments to
+            pass to the task prioritize agent. (default: :obj:`None`)
+        sys_msg_generator_kwargs (Dict, optional): Additional arguments to
+            pass to the system message generator. (default: :obj:`None`)
+        extend_task_specify_meta_dict (Dict, optional): A dict to extend the
+            task specify meta dict with. (default: :obj:`None`)
+        output_language (str, optional): The language to be output by the
+            agents. (default: :obj:`None`)
+        message_window_size (int, optional): The maximum number of previous
+            messages to include in the context window. If `None`, no windowing
+            is performed. (default: :obj:`None`)
+    """
+
+    def __init__(
+        self,
+        assistant_role_name: str,
+        user_role_name: str,
+        task_prompt: str = "",
+        task_type: TaskType = TaskType.AI_SOCIETY,
+        assistant_agent_kwargs: Optional[Dict] = None,
+        task_specify_agent_kwargs: Optional[Dict] = None,
+        task_create_agent_kwargs: Optional[Dict] = None,
+        task_prioritize_agent_kwargs: Optional[Dict] = None,
+        sys_msg_generator_kwargs: Optional[Dict] = None,
+        extend_task_specify_meta_dict: Optional[Dict] = None,
+        output_language: Optional[str] = None,
+        message_window_size: Optional[int] = None,
+    ) -> None:
+        self.task_type = task_type
+        self.task_prompt = task_prompt
+        self.specified_task_prompt: TextPrompt
+        self.init_specified_task_prompt(assistant_role_name, user_role_name,
+                                        task_specify_agent_kwargs,
+                                        extend_task_specify_meta_dict,
+                                        output_language)
+
+        sys_msg_generator = SystemMessageGenerator(
+            task_type=self.task_type, **(sys_msg_generator_kwargs or {}))
+
+        init_assistant_sys_msg = sys_msg_generator.from_dicts(
+            meta_dicts=[dict(task=self.specified_task_prompt)],
+            role_tuples=[
+                (assistant_role_name, RoleType.ASSISTANT),
+            ],
+        )
+
+        self.assistant_agent: ChatAgent
+        self.assistant_sys_msg: BaseMessage
+        self.task_creation_agent: TaskCreationAgent
+        self.task_prioritize_agent: TaskPrioritizeAgent
+        self.init_agents(init_assistant_sys_msg[0], assistant_agent_kwargs,
+                         task_create_agent_kwargs,
+                         task_prioritize_agent_kwargs, output_language,
+                         message_window_size)
+
+        self.tasks: deque = deque([])
+        self.solved_tasks: List[str] = []
+        self.MAX_TASK_HISTORY = 10
```

Comment:
Let's make it an argument
```suggestion
        self.max_task_history = max_task_history
```